### PR TITLE
Replace 'this.resource' with 'this.route'

### PR DESCRIPTION
**Pull request by HeroicEric**
This goes along with https://github.com/ember-cli/ember-cli/pull/3936/files for replacing `this.resource` with `this.route` in generated resource routes.

### CODE REVIEW

**Comment by HeroicEric** [comment]
Does it still make sense to keep `type: 'resource'` around? I think the only difference at this point is that it adds the `function() {}` for you.

File: lib/helpers/resource-node.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ module.exports = function resourceNode(name, options) {
     builders.callExpression(
       builders.memberExpression(
         builders.thisExpression(),
-        builders.identifier('resource'),
+        builders.identifier('route'),
```
Comment (abuiles) [inline lib/helpers/resource-node.js]: We already have a route-node.js which does this, we should probably just get rid of resource-node.js and all the logic related with resource.

File: tests/fixtures/foos-bar-baz-remove-route.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 Router.map(function() {
-  this.resource('foos', function() {
+  this.route('foos', function() {
```
Comment (abuiles) [inline tests/fixtures/foos-bar-baz-remove-route.js]: I think we can remove all files related with resources and have the proper tests for routes

**Comment by abuiles** [comment]
@HeroicEric thanks! I just let some comments, can you please address that ?

**Comment by HeroicEric** [comment]
@abuiles sure.

I was wondering about just removing `resource` too. The only difference that I came across is how it automatically adds the `function() {}` for you like in https://github.com/ember-cli/ember-router-generator/blob/master/tests/generator-test.js#L10-L17 which uses https://github.com/HeroicEric/ember-router-generator/blob/always-use-route/tests/fixtures/foos-resource.js#L2 and

Is it cool if `ember g resource foos` no longer does that?

**Comment by HeroicEric** [comment]
I made the changes you mentioned, removing everything having to do with resources. I'll update the PR for ember-cli

**Comment by abuiles** [comment]
@HeroicEric thanks!

**Comment by grabbou** [comment]
I would suggest printing a note `Using resources is likely to be deprecated in upcoming Ember releases. Please use with caution or switch to using routes.` or just leaving `generate resource` with this note so others know what happened (not everybody is always up to date with the changelog unfortunately)

**Pull request merged**